### PR TITLE
Touch up SSO Config changes

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Some followup work from #17635 

* Updating some copy
* Added clickable actions inside toast notifications for verifying and activating configs. Needed to shift some of the activate & verify logic up a few component levels for this to work right.
* Shifted the `<ConfirmationModal/>`'s `alert` to be above the warning text based on feedback.
* Made the `toast()` function (from `useToast`) accept a ReactNode for the message instead of only a string. This allows for clickable links, etc, inside of toasts. Would be good to iterate here and settle on a pattern for action buttons on toasts to be more explicit. Would also be nice to have a specific error style toast as well. cc @gtsiolis 

| Header | Header |
|--------|--------|
| image | image | 

## Related Issue(s)

Fixes WEB-322

## How to test

* Create an SSO config for your org.
* Try using the verify link in the toast after you save it. You can edit/save the config to get the toast again.
* After verifying, use the Activate link in the toast. You can edit/save/verify to see the toast again if needed.
* Create a new config and use the verify/activate menu items from the list entry with a config as well. You can use the same credentials for the 2nd config w/o any problems.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - bmh-pr-1763f7234a33c</li>
	<li><b>🔗 URL</b> - <a href="https://bmh-pr-1763f7234a33c.preview.gitpod-dev.com/workspaces" target="_blank">bmh-pr-1763f7234a33c.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/dashboard/src/teams/sso/OIDCClients.tsx [hunk 1]
```diff
@@ -29,16 +33,44 @@ type OIDCClientsListProps = {
...
 
     const onCreate = useCallback(() => setShowCreateModal(true), []);
     const hideModal = useCallback(() => setShowCreateModal(false), []);
 
+    const handleSaved = useCallback(
+        (configId: string) => {
+            toast(
+                <span>
+                    Your SSO configuration was successfully saved.{" "}
+                    <LinkButton onClick={() => verifyClient(configId)}>Verify configuration</LinkButton>.
```
Comment (gtsiolis) [inline components/dashboard/src/teams/sso/OIDCClients.tsx:62]: **thought**: Didn't notice we already added a `LinkButton` recently. We could add this as a component variant for the `Button` component but looks good also as is for now—the eternal dilemma: _new component_ vs _new variant_.  😃

File: components/dashboard/src/teams/sso/OIDCClients.tsx [hunk 2]
```diff
@@ -29,16 +33,44 @@ type OIDCClientsListProps = {
...
+    const { toast } = useToast();
     const [showCreateModal, setShowCreateModal] = useState(false);
+    const [activateModalConfigId, setActivateModalConfigID] = useState("");
+
+    const verifyClient = useVerifyClient({
+        onSuccess: (configId: string) => {
+            toast(
+                <span>
+                    Single sign-on configuration was successfully verified.{" "}
+                    <LinkButton onClick={() => setActivateModalConfigID(configId)}>Activate configuration</LinkButton>.
```
Comment (gtsiolis) [inline components/dashboard/src/teams/sso/OIDCClients.tsx:45]: **suggestion**: I'd drop the period from the link button to help it stand out and create better visual balance, as the period looks a bit out of place.

```suggestion
                    <LinkButton onClick={() => setActivateModalConfigID(configId)}>Activate configuration</LinkButton>
```

File: components/dashboard/src/components/ConfirmationModal.tsx [hunk 3]
```diff
@@ -51,14 +51,14 @@ export default function ConfirmationModal(props: {
...
             <ModalBody>
-                <p className="mb-3 text-base text-gray-500">{props.areYouSureText}</p>
                 {props.warningText && (
                     <Alert type="warning" className="mb-4">
                         <strong>{props.warningHead}</strong>
                         {props.warningHead ? ": " : ""}
                         {props.warningText}
                     </Alert>
                 )}
+                <p className="mb-3 text-base text-gray-500">{props.areYouSureText}</p>
```
Comment (gtsiolis) [inline components/dashboard/src/components/ConfirmationModal.tsx:61]: **praise**: Thanks for moving this! 🔮

**Review by gtsiolis — APPROVED** [review]
Great improvements, @selfcontained! 🌮 🌮 

I've opened UX-38 to keep track of the discussions about improving the **toast component**.

Redeploying the preview environment failed (see [relevant job](https://github.com/gitpod-io/gitpod/actions/runs/5007632756/jobs/9006248066?pr=17658)).

UX LGTM from the screenshots, but maybe someone needs to take a closer look at the code changes. 🏁

**Review by AlexTugarev — APPROVED** [review]
That looks great! Thanks for caring so much about the details 🙏🏻 

I recorded a [loom](https://www.loom.com/share/2627ca753d7441b68b391bd88da4105e?highlightComment=12359622&t=23) but that's likely more related to https://github.com/gitpod-io/gitpod/pull/17660 than these changes.

**Comment by selfcontained** [comment]
> That looks great! Thanks for caring so much about the details 🙏🏻
> 
> I recorded a [loom](https://www.loom.com/share/2627ca753d7441b68b391bd88da4105e?highlightComment=12359622&t=23) but that's likely more related to #17660 than these changes.

Yup, that's exactly what that other PR fixes.

**Pull request merged**